From 295b423995edb1f0f9b954b042ee382eeda23822 Mon Sep 17 00:00:00 2001
From: Don Darling <don.osc2@gmail.com>
Date: Tue, 13 Jul 2010 19:30:51 -0500
Subject: [PATCH 04/14] v4l2src: make sure capture buffer size is aligned on DM365.

On DM365, capture buffers use a pitch that is aligned to a 32-byte boundary.
We need to comprehend this when calculated the expected capture buffer size.
---
 sys/v4l2/gstv4l2object.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index f65c458..d7f0d6a 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -1204,8 +1204,14 @@ gst_v4l2_object_get_caps_info (GstV4l2Object * v4l2object, GstCaps * caps,
         break;
       case GST_MAKE_FOURCC ('N', 'V', '1', '2'):
         fourcc = V4L2_PIX_FMT_NV12;
+#if defined(Platform_dm365)
+        /* NV12 buffer on dm365 is 32-byte aligned */
+        outsize = GST_ROUND_UP_32 (*w) * GST_ROUND_UP_2 (*h);
+        outsize += (GST_ROUND_UP_32 (*w) * *h) / 2;
+#else
         outsize = GST_ROUND_UP_4 (*w) * GST_ROUND_UP_2 (*h);
         outsize += (GST_ROUND_UP_4 (*w) * *h) / 2;
+#endif
         break;
       case GST_MAKE_FOURCC ('N', 'V', '2', '1'):
         fourcc = V4L2_PIX_FMT_NV21;
-- 
1.7.0.4


